File: 240201/길이가 무한대인 수평선 트랙/horizontal-track-with-infinite-length.py
N, T = map(int, input().split())

arr = []

for _ in range(N):
    start, speed = map(int, input().split())
    end = speed*T+start
    arr.append((start, end))
arr.sort()

temp = arr.pop()[1]
answer = 1
while arr:
    _, end = arr.pop()
    if temp > end:
        answer += 1
        temp = end
print(answer)

File: 240201/길이가 무한대인 수평선 트랙/horizontal-track-with-infinite-length.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    int answer = 1;
    int N, T;
    vector<pair<int, int>> v;


    int start, speed, end;
    cin >> N >> T;
    
    for (int i=0;i<N;i++){
        cin >> start >> speed;
        end = start + speed * T;
        v.push_back(pair<int, int>(start, end));
    }

    sort(v.begin(), v.end());
    reverse(v.begin(), v.end());
    
    int temp = v[0].second;

    for (int i=1;i<N;i++)
        if (temp > v[i].second){
            answer++;
            temp = v[i].second;
        }
    cout << answer;
    return 0;
}